zpay32/amountunits.go

```diff
@@ -127,6 +127,11 @@ func decodeAmount(amount string) (lnwire.MilliSatoshi, error) {
 // encodeAmount encodes the provided millisatoshi amount using as few characters
 // as possible.
 func encodeAmount(msat lnwire.MilliSatoshi) (string, error) {
+	if msat < 0 {
+		return "", fmt.Errorf("unable to encode negative amount: %v",
```

Comment:
It should say that negative amounts are not allowed. Maybe "amount must be positive: %v"? :)